src/opm/output/eclipse/LoadRestart.cpp

```diff
@@ -76,79 +107,92 @@ class RestartFileView
         return this->sim_step_;
     }
 
-    operator const Opm::RestartIO::ecl_file_view_type*() const
+    template <typename ElmType>
+    bool hasKeyword(const std::string& vector) const
```

Comment:
The different types - INT, FLOAT, ... are not in separate namespaces in the file, i.e. the `hasKeyword()` method can (and should?) be independent of `ElmType`

Reply:
> the `hasKeyword()` method can (and should?) be independent of `ElmType`

Only if we accept an order of magnitude slowdown in its response time.

Reply:
Ehhh - this should be some map like structure - i.e. `log( )` lookup time.

Reply:
> Ehhh - this should be some map like structure - i.e. `log( )` lookup time.

Then we either end up storing the strings twice in `RestartFileView` because class `ERst` does not offer that operation out of the box or add an `ERst::hasKey()` member function that's restricted to a single sequence number. On a side note I also think there is benefit to knowing that  the result set's `IWEL` vector actually has the expected type.

Reply:
ok